test/error_messages.coffee

```diff
@@ -989,3 +991,39 @@ test "`&&=` and `||=` with a space in-between", ->
     a or = 1
          ^
   '''
+
+test "anonymous functions cannot be exported", ->
+  assertErrorFormat '''
+    export ->
+      console.log 'hello, world!'
+  ''', '''
+    [stdin]:1:8: error: unexpected ->
+    export ->
+           ^^
+  '''
+
+test "anonymous classes cannot be exported", ->
+  assertErrorFormat '''
+    export class
+      @constructor: ->
```

Comment:
Do you really mean to include that `@` here? It sure is valid, but it's some very odd code :)